### ISSUE TITLE
Incorrect random generators

### ISSUE THREAD

**Issue by BeardOfDoom** [cryptid-org/cryptid-native#27]
I checked the random generators in [Random.c](https://github.com/cryptid-org/cryptid-native/blob/master/src/util/Random.c) and I think the `random_unsignedIntInRange` and the `random_mpzInRange` functions might be incorrect also all the functions are poorly documented in this file.

Further investigation required and documentation for the random generators.

**Comment by battila7** [comment]
Nice catch! Replaced the invalid tag with bug, since this can very well be called a bug.

**Comment by battila7** [comment]
Documentation and reimplementation can be postponed until #21 is resolved.

**Comment by battila7** [comment]
Change of plans: Implementation should be done prior to the resolution of #21. Fixed random generation should be part of v1

**Comment by battila7** [comment]
Closing as #34 is merged.
